src/util/testing.ts

```diff
@@ -0,0 +1,73 @@
+import { serverConfig } from '../config'
+import { slackStatus } from './services/slackAlert'
+import { datelog } from './utils'
+
+interface TestResult {
+  description: string
+  pass: boolean
+  error?: string
+}
+
+let results: TestResult[] = []
+
+export const describe = async (
+  description: string,
+  func: () => Promise<void>
+): Promise<void> => {
+  datelog(`*** ${description} START ***`)
+  results = []
+  await func()
+  datelog(`*** ${description} END *****\n`)
+  const failures = results.filter(r => !r.pass)
+  if (failures.length > 0) {
+    const messages = failures.map(
+      f => `[FAILED] ${f.description}: ${f.error ?? ''}`
+    )
+    const message = `FAILED Test group ${description}\n` + messages.join('\n')
+    console.log(`Messaging Errors to Slack\n${message}`)
+    slackStatus(message)
+  }
+}
```

Comment:
If we are re-building mocha, maybe we should drop the dependency from our `package.json`.

Also, `describe` is supposed to take a plain function that returns `void`, not a promise. Calling `it` inside this `describe` body will queue up a test function in an array. Finally, once the `describe` body is done running, we go through the test functions in a loop and run them.

Otherwise, your version needs an `await it()` syntax, which differs from the way `it` works in Jest or Mocha. If you don't want to do a bunch of extra work to make these things more similar, I guess we can live with this difference.

But actually, why re-invent Mocha in the first place? Mocha can already load files from a folder (so you can drop your plugins in there), and it provides `beforeEach` / `afterEach` hooks that could be used for your logging needs.

Reply:
Lets sidebar on this, the thought process on how to structure the tests went through a few iterations before this PR and it's retaining some considerations that I haven't fully utilized.. yet. I'd love your feedback on this.

Reply:
1. Not using mocha directly as I don't know of a way to run mocha as a long running nodejs task that loops indefinitely. It's unfortunately not a library that can run within node but it's instead an executable itself.
2. Yeah, I don't care too much to make them super similar. I can migrate to that later if we feel the need.
3. We are pulling the fixtures from the database and all using one simple plugin which only knows how to fetch and check results match a fixture. In the future we'll have other plugins that can do more complex logic and may have many describe/it blocks and uses fewer data from the fixtures itself. We now have that freedom.